test/test_document.rb

```diff
@@ -44,6 +44,16 @@ def assert_equal_value(key, one, other)
       assert_equal "foo.bar", @document.data["whatever"]
     end
 
+    should "be able to jsonify its data" do
+      require 'json'
```

Comment:
Ah, I misspoke. It's required in `#inspect`. So it looks like you'll have to manually require it either in `lib/jekyll.rb` or as the first line of the body of this method.

Why don't you just omit certain keys? Maybe something like:

```ruby
keys.each_with_object({}) do |(key, _), result|
  result[key] = self[key] unless %w{previous next}.include? key
end
```